packages/mdl-drawer/temporary/index.js

```diff
@@ -0,0 +1,138 @@
+      const el = document.createElement('div');
+      this.transformPropertyName_ = ('transform' in el.style ? 'transform' : '-webkit-transform');
+    }
+    return this.transformPropertyName_;
+  }
+
+  static get supportsCSSCustomProperties() {
+    if (this.supportsCSSCustomProperties_ === undefined) {
+      this.supportsCSSCustomProperties_ = CSS.supports('(--color: red)');
+    }
+    return this.supportsCSSCustomProperties_;
+  }
+
+  get open() {
+    return this.foundation_.isOpen();
+  }
+
+  set open(value) {
+    if (value) {
+      this.foundation_.open();
+    } else {
+      this.foundation_.close();
+    }
+  }
+
+  /* Return the drawer element inside the component. */
+  get drawer() {
+    return this.root_.querySelector(MDLTemporaryDrawerFoundation.strings.DRAWER_SELECTOR);
+  }
+
+  getDefaultFoundation() {
+    const {FOCUSABLE_ELEMENTS, OPACITY_VAR_NAME, TAB_DATA, TAB_DATA_HANDLED} = MDLTemporaryDrawerFoundation.strings;
+
+    return new MDLTemporaryDrawerFoundation({
+      addClass: className => this.root_.classList.add(className),
+      removeClass: className => this.root_.classList.remove(className),
+      hasClass: className => this.root_.classList.contains(className),
+      hasNecessaryDOM: () => Boolean(this.drawer),
+      registerComponentInteractionHandler: (evt, handler) =>
+          this.root_.addEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      deregisterComponentInteractionHandler: (evt, handler) =>
+          this.root_.removeEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      registerDrawerInteractionHandler: (evt, handler) =>
+          this.drawer.addEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      deregisterDrawerInteractionHandler: (evt, handler) =>
+          this.drawer.removeEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      registerTransitionEndHandler: handler => this.drawer.addEventListener('transitionend', handler),
+      deregisterTransitionEndHandler: handler => this.drawer.removeEventListener('transitionend', handler),
+      getDrawerWidth: () => this.drawer.offsetWidth,
+      updateDrawerPosition: value => this.drawer.style.setProperty(
+          MDLTemporaryDrawer.transformPropertyName, `translateX(${value}px)`),
+      updateComponentBackground: value => {
+        if (MDLTemporaryDrawer.supportsCSSCustomProperties) {
+          this.root_.style.setProperty(OPACITY_VAR_NAME, value);
+        }
+      },
+      getFocusableElements: () => Array.from(this.drawer.querySelectorAll(FOCUSABLE_ELEMENTS)),
+      saveElementTabState: el => {
+        if (el.hasAttribute('tabindex')) {
+          el.setAttribute(TAB_DATA, el.getAttribute('tabindex'));
+        }
+        el.setAttribute(TAB_DATA_HANDLED, true);
+      },
+      restoreElementTabState: el => {
+        // Only modify elements we've already handled, in case anything was dynamically added since we saved state.
+        if (el.hasAttribute(TAB_DATA_HANDLED)) {
```

Comment:
Oh, whoops! Forgot about this one. Moved it and `saveElementTabState` too, for symmetry.